udsoncan/connections.py

```diff
@@ -735,15 +735,24 @@ def __init__(self, windll: str, rxid: int, txid: int, name: Optional[str] = None
         txFlags = TxStatusFlag.ISO15765_CAN_ID_29.value if txid >> 11 else TxStatusFlag.ISO15765_CAN_ID_11.value
 
         # Set up a J2534 interface using the DLL provided
-        self.interface = J2534(windll=windll, rxid=rxid, txid=txid, txFlags=txFlags)
+        try:
+            self.interface = J2534(windll=windll, rxid=rxid, txid=txid, txFlags=txFlags)
+        except FileNotFoundError:
+            raise RuntimeError('DLL not found')
 
         # Set the protocol to ISO15765, Baud rate to 500000
         self.protocol = Protocol_ID.ISO15765
         self.baudrate = 500000
         self.debug = debug
 
-        # Open the interface (connect to the DLL)
-        result, self.devID = self.interface.PassThruOpen()
+        try:
+            # Open the interface (connect to the DLL)
+            self.result, self.devID = self.interface.PassThruOpen()
+            self.log_last_operation("PassThruOpen", with_raise=True)
+        except WindowsError as e:

```

Comment:
Hmm this new exception causes CI to fail because CI runs on Linux and this is unknown.
We would probably need a more complex config that sets the OS to win32 when analyzing this file.  I think this is doable with a .mypy.ini